src/distilabel/steps/task/data.py

```diff
@@ -0,0 +1,81 @@
+# Copyright 2023-present, Argilla, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import TYPE_CHECKING, Any, Dict, List, Optional
+
+from typing_extensions import override
+
+from distilabel.llm.base import LLM
+from distilabel.steps.base import (
+    GeneratorStep,
+)
+from distilabel.steps.task.base import _Task
+
+if TYPE_CHECKING:
+    from distilabel.steps.task.typing import ChatType
+    from distilabel.steps.typing import GeneratorStepOutput
+
+
+class DataTask(_Task, GeneratorStep):
+    """DataTask is a class that implements the `_Task` abstract class and adds the
+    `GeneratorStep` interface to be used as a step in the pipeline.
+
+    Args:
+        data: The data to be used to generate the outputs of the task.
+    """
+
+    llm: LLM = None
+    group_generations: Optional[bool] = None
+    num_generations: Optional[int] = None
+    generation_kwargs: Optional[dict] = None
+    data: List[Dict[str, Any]]
+
+    @override
+    def load(self) -> None:
+        pass
+
+    @override
+    def process(self) -> "GeneratorStepOutput":  # type: ignore
```

Comment:
`GeneratorStep` also needs to receive an `offset: int = 0` parameter. This is mainly for the caching, so if a pipeline is loaded from cache the generator nows where it stopped in the previous run and from where it has to restart yielding batches.